flow360/component/simulation/models/surface_models.py

```diff
@@ -0,0 +1,128 @@
+"""
+Contains basically only boundary conditons for now. In future we can add new models like 2D equations.
+"""
+
+from abc import ABCMeta
+from typing import List, Literal, Optional, Tuple, Union
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+
+from .turbulence_quantities import TurbulenceQuantitiesType
+
+BoundaryVelocityType = Unioin[VelocityType.Vector, Tuple[pd.StrictStr, pd.StrictStr, pd.StrictStr]]
+
+
+class BoundaryBase(Flow360BaseModel, metaclass=ABCMeta):
+    type: str = pd.Field()
+    surfaces: EntityList[Surface] = pd.Field()
+
+
+class BoundaryBaseWithTurbulenceQuantities(BoundaryBase, metaclass=ABCMeta):
+    turbulence_quantities: Optional[TurbulenceQuantitiesType] = pd.Field(None)
+
+
+class Wall(BoundaryBase):
+    """Replace Flow360Param:
+    - NoSlipWall
+    - IsothermalWall
+    - HeatFluxWall
+    - WallFunction
+    - SolidIsothermalWall
+    - SolidAdiabaticWall
+    """
+
+    model_config = ConfigDict(
+        conflicting_fields=[Conflicts("temperature", "heat_flux")],
+    )
+
+    type: Literal["Wall"] = pd.Field("Wall", frozen=True)
+    use_wall_function: bool = pd.Field(False)
+    velocity: Optional[BoundaryVelocityType] = pd.Field(None)
+    velocity_type: Literal["absolute", "relative"] = pd.Field("relative")
+    temperature: Optional[Union[pd.PositiveFloat, pd.StrictStr]] = pd.Field()
+    heat_flux: Optional[Union[float, pd.StrictStr]] = pd.Field(None)
+
+
+class VelocityFromMachAndAngles(Flow360BaseModel):
+    Mach: float = pd.Field()
+    beta: float = pd.Field(0)
+    alpha: float = pd.Field(0)
+
+
+class VelocityFromMagnitudeAndAngles(Flow360BaseModel):
+    velocity: VelocityType = pd.Field()
+    beta: float = pd.Field(0)
+    alpha: float = pd.Field(0)
+
+
+class Freestream(BoundaryBaseWithTurbulenceQuantities):
+    type: Literal["Freestream"] = pd.Field("Freestream", frozen=True)
+    surfaces: Optional[EntityList[Surface]] = pd.Field(None)
+    velocity: Optional[
+        Union[VelocityFromMachAndAngles, VelocityFromMagnitudeAndAngles, BoundaryVelocityType]
+    ] = pd.Field(None)
+    velocity_type: Literal["absolute", "relative"] = pd.Field("relative")
+
+
+class Outflow(BoundaryBaseWithTurbulenceQuantities):
+    """Replace Flow360Param:
+    - SubsonicOutflowPressure
+    - SubsonicOutflowMach
+    - MassOutflow
+    """
+
+    type: Literal["Outflow"] = pd.Field("Outflow", frozen=True)
+    pressure: Optional[PressureType] = pd.Field(None)
+    Mach: Optional[float] = pd.Field(None)
```

Comment:
what if user wants to use velocity?

Reply:
That's not implemented in our solver

Reply:
Our solver currently only supports user supplied pressure or Mach number at the outflow boundary. The state here is not fully known, so specifying Mach number is not the same as specifying velocity

Reply:
you mean boundary condition is on enforcing local Mach number which depends on local temperature. How users come up with Mach number for the outflow?

Reply:
Let's keep as is for now.